src/main/java/cloud/orbit/actors/cluster/impl/RedisConnectionManager.java

```diff
@@ -140,22 +140,34 @@ public void subscribeToChannel(final String channelId, final MessageListener<Obj
     public void sendMessageToChannel(final String channelId, final Object msg)
     {
             final List<RedisOrbitClient> localMessagingClients = messagingClients.stream().filter((e) -> e.isConnectied()).collect(Collectors.toList());
-            final int activeClientCount = localMessagingClients.size();
-            if(activeClientCount > 0)
-            {
-                final int randomId = ThreadLocalRandom.current().nextInt(activeClientCount);
-                final RedissonClient client = localMessagingClients.get(randomId).getRedissonClient();
-
-                client.getTopic(channelId).publishAsync(msg).exceptionally((e) ->
-                {
-                    logger.error("Error sending message", e);
-                    return 0L;
-                });
-            }
-            else
-            {
-                throw new UncheckedException("No Redis messaging instances available.");
-            }
+            sendMessageToChannel(channelId, msg, localMessagingClients, 0);
+    }
+
+    private void sendMessageToChannel(final String channelId, final Object msg, final List<RedisOrbitClient> localMessagingClients, final int attempt) {
+        final int activeClientCount = localMessagingClients.size();
+        if(activeClientCount > 0)
+        {
+            final int randomId = ThreadLocalRandom.current().nextInt(activeClientCount);
+            final RedissonClient client = localMessagingClients.get(randomId).getRedissonClient();
+
+            client.getTopic(channelId).publishAsync(msg)
+                    .whenComplete((x, e) -> {
+                       if(e != null) {
+                           logger.error("Failed to send message to channel '{}'", channelId, e);
+                       } else if(x == 0) {
+                           if(attempt >= redisClusterConfig.getMessageResendAttempts()) {
+                               logger.error("Failed to send message to channel '{}' after {} attempts.", channelId, attempt);
+                           } else {
+                               logger.warn("Failed to send message to channel '{}' on attempt {}. Retrying...", channelId, attempt);
```

Comment:
Worth including the `client` identifier of some sort. Host name or otherwise so that if we see a bunch of these messages and they are all for a specific client we can correlate that.

Reply:
The "channel" is the identifier for the remote node you are trying to contact. We don't have any info beyond that at this stage.

Reply:
The "messaging node / client" isn't the same as the "channel" for the backend host. What would be good to know is if a node wasn't healthy or unavailable so a bunch of backend hosts couldn't create their "channel" then we could see that with the client identifier.

Reply:
I don't think we have that info at this point, so passing that down would be a much deeper change.
We can potentially open that as another ask but I think it's outside the scope of this change.